virtualizarr/writers/icechunk.py

```diff
@@ -188,6 +217,49 @@ def virtual_datatree_to_icechunk(
         )
 
 
+def validate_virtual_chunk_containers(
```

Comment:
I kinda feel like I should be able to just call some public Icechunk API here to do this instead of doing it myself. Like there is currently no way to validate a virtual reference without then immediately setting it.

Reply:
Raised https://github.com/earth-mover/icechunk/issues/1167#issue-3319420773 upstream to track this idea.